src/fr/mrc/ptichat/main/ui/ConnexionUI.java

```diff
@@ -0,0 +1,121 @@
+package fr.mrc.ptichat.main.ui;
+
+import fr.mrc.ptichat.main.utils.LanguagesController;
+
+import javax.swing.*;
+import javax.swing.border.EmptyBorder;
+import java.awt.*;
+
+public class ConnexionUI extends JFrame {
+
+    private LanguagesController languagesController = new LanguagesController("English");
+
+    public ConnexionUI() {
+        this.setGlobalParameters();
+        this.createUI();
+        this.pack();
+    }
+
+    public void open(){
+        this.setVisible(true);
+    }
+
+    /**
+     * Formats the connexion window.
+     */
+    private void setGlobalParameters(){
+        this.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        this.setMinimumSize(new Dimension(500, 100));
+        this.setResizable(false);
+        this.setLocationRelativeTo(null);
+    }
+
+    /**
+     * Creates the global architecture of the window.
+     * Divides the window into three rows, containing respectively the title, the form and the validation button.
+     */
+    private void createUI() {
+        this.setLayout(new BorderLayout(10, 10));
+        JPanel titlePanel = new JPanel();
+        JPanel formPanel = new JPanel();
+        JPanel validationPanel = new JPanel();
+        titlePanel.setBorder(new EmptyBorder(10, 10, 10, 10));
+        formPanel.setBorder(new EmptyBorder(10, 10, 10, 10));
+        validationPanel.setBorder(new EmptyBorder(10, 10, 10, 10));
+        this.initTitlePanel(titlePanel);
+        this.initConnexionForm(formPanel);
+        this.initValidationPanel(validationPanel);
+        this.getContentPane().add(titlePanel, BorderLayout.NORTH);
+        this.getContentPane().add(formPanel, BorderLayout.CENTER);
+        this.getContentPane().add(validationPanel, BorderLayout.SOUTH);
+    }
+
+    /**
+     * Initializes the title panel.
+     * @param container the <code>Container</code> to fulfill
+     */
+    private void initTitlePanel(Container container) {
+        JLabel title = new JLabel();
+        title.setText(this.languagesController.getWord("WELCOME"));
+        container.add(title);
+    }
+
+    /**
+     * Initializes the form panel.
+     * @param container the <code>Container</code> to fulfill
+     */
+    private void initConnexionForm(Container container) {
```

Comment:
Why don't you manipulate a String[]{"USER_IP", "USER_PORT", "PEER_IP", "PEER_PORT"} to generate an array of JLabels and an array of JPanels. You could then add all these to the view with a for loop :)